tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/CalendarEventReplyGenerator.scala

```diff
@@ -3,42 +3,56 @@ package com.linagora.tmail.james.jmap.model
 import java.net.URI
 
 import com.google.common.base.Preconditions
-import net.fortuna.ical4j.model.component.VEvent
-import net.fortuna.ical4j.model.parameter.{CuType, PartStat, Role}
+import net.fortuna.ical4j.model.component.{VEvent, VTimeZone}
+import net.fortuna.ical4j.model.parameter.{Cn, CuType, PartStat, Role}
 import net.fortuna.ical4j.model.property.{Attendee, Method}
 import net.fortuna.ical4j.model.{Calendar, Property, PropertyList}
 import org.apache.james.core.Username
 
 import scala.jdk.CollectionConverters._
 
+case class AttendeeReply(attendee: Username, partStat: PartStat, cn : Option[Cn] = None)
+
 object CalendarEventReplyGenerator {
-  def generate(eventRequest: Calendar, attendee: Username, partStat: PartStat): Calendar = {
+
+  val PRODID: String = "-//Linagora//TMail Calendar//EN"
+
+  def generate(eventRequest: Calendar, attendeeReply: AttendeeReply): Calendar = {
     Preconditions.checkArgument(eventRequest.getMethod.equals(Method.REQUEST), "The method of the event request must be REQUEST".asInstanceOf[Object])
-    val requestVEvent: VEvent = eventRequest.getComponents("VEVENT").asInstanceOf[java.util.List[VEvent]].asScala.head
+    val requestVEventOpt: Option[VEvent] = eventRequest.getComponents[VEvent]("VEVENT").asScala.headOption
+    Preconditions.checkArgument(requestVEventOpt.isDefined, "The event request must be a VEVENT".asInstanceOf[Object])
+
+    val requestVEvent: VEvent = requestVEventOpt.get
+    val timeZone: Option[VTimeZone] = requestVEvent.getComponents[VTimeZone]("VTIMEZONE").asScala.headOption
 
     val replyProperty: PropertyList[Property] = Some(new PropertyList[Property])
       .map(propertyList => {
         requestVEvent.getProperties
           .stream()
           .filter(!_.getName.equals("ATTENDEE"))
           .forEach(e => propertyList.add(e))
-        propertyList.add(buildReplyAttendee(attendee, partStat))
+        propertyList.add(buildReplyAttendee(attendeeReply))
         propertyList
       }).get
 
     val calendar = new Calendar()
       .withDefaults()
       .withComponent(new VEvent(replyProperty).getFluentTarget)
+      .withProdId(PRODID)
```

Comment:
Is the prodid necessary? TMail calendar does not really exist as well right?^^'

Reply:
We likely need to override the sender PROID though (origin e.g. Google Calendar). Not sure if we can let it empty though.

Maybe keep Sabre or somethign currently as OpenPaaS calendar?